manifests/init.pp

```diff
@@ -53,8 +57,10 @@
 #
 class hash_resources(
   $resources = {},
+  $realize = {},
```

Comment:
Not sure but it could be as simple as changing datatypes and just passing:
realize:
 - File[/tmp/myfile]
 - Ssh_authorized_key[root]

will try it ...